pkg/agent/multicluster/mc_route_controller.go

```diff
@@ -436,10 +625,31 @@ func generatePeerConfigs(subnets []string, gatewayIP net.IP) (map[*net.IPNet]net
 	return peerConfigs, nil
 }
 
-// getPeerGatewayIP will always return the first Gateway IP.
-func getPeerGatewayIP(spec mcv1alpha1.ClusterInfo) net.IP {
+// getPeerGatewayIP will return the network IP address of the Service CIDR when WireGuard enabled,
+// or it will return the first IP of the Gateway.
+func getPeerGatewayIP(spec mcv1alpha1.ClusterInfo, enableWireGuard bool) net.IP {
+	if enableWireGuard {
+		_, serviceCIDR, err := net.ParseCIDR(spec.ServiceCIDR)
+		if err != nil {
+			klog.Warning("Failed to get the peer Gateway IP for WireGuard, return nil")
```

Comment:
You should add returned err to the warning log.

Reply:
Added.